azurerm/resource_arm_application_gateway.go

```diff
@@ -1527,6 +1533,11 @@ func expandApplicationGatewayBackendHTTPSettings(d *schema.ResourceData, gateway
 			setting.ApplicationGatewayBackendHTTPSettingsPropertiesFormat.HostName = utils.String(hostName)
 		}
 
+		affinityCookieName := v["affinity_cookie_name"].(string)
+		if affinityCookieName != "" {
+			setting.AffinityCookieName = utils.String(affinityCookieName)
```

Comment:
do we not always want to assign this to ensure the value can be removed?

Reply:
It looks like it gets removed when you don't pass it in so we're good the way it is!